File: frontend/lib/types/deck.ts
export interface DeckDTO {
  id: number;
  name: string;
  description: string;
  createdAt: string;
}

export interface CreateDeckRequest {
  name: string;
  description: string;
}

export interface UpdateDeckRequest {
  name: string;
  description: string;
}

File: backend/src/main/java/com/peanuts/anki/deck/Deck.java
package com.peanuts.anki.deck;

import com.peanuts.anki.auth.User;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "decks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Deck {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(length = 1000)
    private String description;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User owner;

    private LocalDateTime createdAt = LocalDateTime.now();
}